Add missing keywords from PHP

mcfedr opened the pull request.

pyrmont commented:
@mcfedr I'm sorry it's taken so long to get back to you on this :(

This PR looks really good. Thank you especially for adding tests to the spec. Would you also be able to add a couple of examples using the new keywords to the visual sample? It's in `spec/visual/samples/php`. That would be very helpful!

mcfedr commented:
Hi @pyrmont It has been a little while :)

I've added a few more examples, and actually a couple more cases - there is still work to be done to make the samples look as good as they could but I've not got the time right now to work on it.

Also rebased on master, and run the command `rake builtins:php` to update the builtins as a couple of php releases have changed them a bit.

pyrmont commented on spec/visual/samples/php:
```diff
@@ -291,11 +374,11 @@ class Zip extends Archive {
   }
-  
+
  /**
```
I realise that this isn't actually from your code but this looks like it's rendering incorrectly in the visual test app. It's supposed to be a multiline comment, right?

mcfedr replied:
These are just trailing whitespaces that I've accidentally removed with my editor, seem to be rendered just fine though.

pyrmont replied:
Sorry, my comment wasn't very clear. In the visual sample there is this text:

```
 /**
 *  Zip class file
 *
 *  @package     fnord.bb
 *  @subpackage  archive
 */
```

It's being rendered as a string but it should be a comment, right?

Just to be clear, your code hasn't caused this error (if it is an error); it's there right now. I was just taking advantage of asking you since you seemed more knowledgeable about PHP than me :D

mcfedr replied:
ah, I see what you mean, I just saw it was marked up in a colour and assumed it was the right one, yes you are right.

mcfedr replied:
Fixed this with ca1d28c

pyrmont commented on spec/visual/samples/php:
```diff
@@ -27,6 +27,89 @@ echo $ΔΔ;
+use const some\namespace\ConstC;
+
+// Grouped uses
+use some\namespace\{ClassA, ClassB, ClassC as C};
```
The second slash here is being rendered as an error token. Presumably that's not what you intended, right?

mcfedr replied:
This is one of the cases where I haven't finished the job, it needs more work to render this correctly, but as this is valid php code I've added the example.

mcfedr replied:
Just fixed this up.

pyrmont commented:
@mcfedr Rouge is a better library with a more up to date version of the PHP lexer (and good to fix a tokenisation bug, too!). Thanks for all your help!

We currently have a two-week cadence for minor releases. As it happens, 3.6.0 went out yesterday and so if everything sticks to schedule, this should be available as part of 3.7.0 on Tuesday 23 July.

mcfedr commented:
Awesome, great to see this merged!
